teloc/src/container.rs

```diff
@@ -61,36 +61,48 @@ impl<T> Container for SingletonContainer<T> {
         Self(OnceCell::new())
     }
 }
-impl<T, Deps> ResolveContainer<'_, T, Self, Deps> for SingletonContainer<T>
+impl<'a, T, Deps> ResolveContainer<'a, &'a T, Self, Deps> for SingletonContainer<T>
 where
-    T: Dependency<Deps> + DependencyClone,
+    T: Dependency<Deps> + 'a,
 {
-    fn resolve_container<F: Fn() -> Deps>(ct: &Self, get_deps: F) -> T {
+    fn resolve_container<F: Fn() -> Deps>(ct: &'a Self, get_deps: F) -> &'a T {
         let elem_ref = ct.get().get();
         match elem_ref {
             None => {
                 let needed = get_deps();
                 let dep = T::init(needed);
-                match ct.get().set(dep.clone()) {
+                match ct.get().set(dep) {
                     Ok(()) => {}
                     Err(_) => unreachable!("Should never been reached"),
                 }
-                dep
+                ct.get().get().expect("Should never been failed")
```

Comment:
I always hate when I need to write something like `var.get().get().get()`. This is not informative because intermediate types are erased. Maybe you can propose a way to remove it?

Reply:
I actually fixed this in my last commit. I was able to replace this whole block of code with a pre-existing method call.